geomet/wkt.py

```diff
@@ -195,7 +195,11 @@ def _round_and_pad(value, decimals):
         # if get a `decimals` value of 0, we want to return an int.
         return repr(int(round(value, decimals)))
 
-    rounded = repr(round(value, decimals))
+    if repr(value).find('e-') == 1:
```

Comment:
It's also valid for number representations to have a positive exponent. For example:

```
>>> repr(1e16)
'1e+16'
```

Furthermore, the expected index of the `find()` expression here is not going to work in all cases. For example:

```
>>> repr(3.14e-16).find('e-')
4
>>> repr(3.14e16).find('e-')
-1
```

I would suggest simply replacing line 198 with `if 'e' in repr(value):`.